src/Compilers/CSharp/Test/Syntax/Parsing/LocalFunctionParsingTests.cs

```diff
@@ -13,6 +13,92 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+        public void IncompleteLocalFunc()
+        {
+            var file = ParseFile(@"
+class C
+{
+    void M1()
+    {
+        await L<
+    }
+    void M2()
+    {
+        int L<
+    }
+    void M3()
+    {
+        int? L<
+    }
+    void M4()
+    {
+        await L(
+    }
+    void M5()
+    {
+        int L(
+    }
+    void M6()
+    {
+        int? L(
+    }
+}");
+            Assert.NotNull(file);
+
+            var m1 = file.DescendantNodes()
+                .OfType<MethodDeclarationSyntax>()
+                .Where(m => (string)m.Identifier.Value == "M1")
+                .Single();
+            var m1Stmts = m1.Body.Statements;
+            Assert.Equal(2, m1Stmts.Count);
+            Assert.Equal(SyntaxKind.LocalDeclarationStatement, m1Stmts[0].Kind());
+            Assert.Equal("await L", m1Stmts[0].ToFullString().Trim());
+            Assert.Equal(SyntaxKind.ExpressionStatement, m1Stmts[1].Kind());
+            Assert.Equal("<", m1Stmts[1].ToFullString().Trim());
+
+            var m2 = file.DescendantNodes()
+                .OfType<MethodDeclarationSyntax>()
+                .Where(m => (string)m.Identifier.Value == "M2")
+                .Single();
+            var m2s1 = m2.Body.Statements.Single();
+            Assert.Equal(SyntaxKind.LocalFunctionStatement, m2s1.Kind());
+            Assert.Equal("int L<", m2s1.ToFullString().Trim());
+
+            var m3 = file.DescendantNodes()
+                .OfType<MethodDeclarationSyntax>()
+                .Where(m => (string)m.Identifier.Value == "M3")
+                .Single();
+            var m3s1 = m3.Body.Statements.Single();
+            Assert.Equal(SyntaxKind.LocalFunctionStatement, m3s1.Kind());
+            Assert.Equal("int? L<", m3s1.ToFullString().Trim());
+
+            var m4 = file.DescendantNodes()
+                .OfType<MethodDeclarationSyntax>()
+                .Where(m => (string)m.Identifier.Value == "M4")
+                .Single();
+            var m4s1 = m4.Body.Statements.Single();
+            Assert.Equal(SyntaxKind.LocalDeclarationStatement, m4s1.Kind());
+            Assert.Equal("await L(", m4s1.ToFullString().Trim());
+
+            var m5 = file.DescendantNodes()
+                .OfType<MethodDeclarationSyntax>()
+                .Where(m => (string)m.Identifier.Value == "M5")
+                .Single();
+            var m5s1 = m5.Body.Statements.Single();
+            Assert.Equal(SyntaxKind.LocalFunctionStatement, m5s1.Kind());
+            Assert.Equal("int L(", m5s1.ToFullString().Trim());
+
+            var m6 = file.DescendantNodes()
+                .OfType<MethodDeclarationSyntax>()
+                .Where(m => (string)m.Identifier.Value == "M6")
+                .Single();
+            var m6s1 = m6.Body.Statements.Single();
+            Assert.Equal(SyntaxKind.LocalFunctionStatement, m6s1.Kind());
+            Assert.Equal("int? L(", m6s1.ToFullString().Trim());
+        }
```

Comment:
If this is verifying the shape of the syntax tree, consider using the helpers in `ParsingTests`. See `AsyncParsingTests` for example.